egret/models/dcopf.py

```diff
@@ -125,12 +125,113 @@ def create_btheta_dcopf_model(model_data):
                                                  )
 
     ### declare angle difference limits on interconnected buses
-    libbranch.declare_ineq_angle_diff_branch_lbub(model=model,
+    if include_angle_diff_limits:
+        libbranch.declare_ineq_angle_diff_branch_lbub(model=model,
+                                                      index_set=branch_attrs['names'],
+                                                      branches=branches,
+                                                      coordinate_type=CoordinateType.POLAR
+                                                      )
+
+    ### declare the generator cost objective
+    libgen.declare_expression_pgqg_operating_cost(model=model,
+                                                  index_set=gen_attrs['names'],
+                                                  p_costs=gen_attrs['p_cost']
+                                                  )
+
+    obj_expr = sum(model.pg_operating_cost[gen_name] for gen_name in model.pg_operating_cost)
+
+    model.obj = pe.Objective(expr=obj_expr)
```

Comment:
Here and below - I was expecting to see slack variables in the objective, appropriately penalized.

Reply:
I've updated the code to enable kwargs input into the model build, along with logic that will appropriately build a penalty if slacks are included

Reply:
Looks good!